src/Voron/Data/Containers/Container.cs

```diff
@@ -441,32 +448,27 @@ private static Container MoveToNextPage(LowLevelTransaction llt, long containerI
             // This is the case where at some point we need to just give up or end up wasting more time to find a page than the time
             // we will use to create and store in disk a new one.
             int i = 0;
+            using var it = ScanFreePages(GetFreeListTree(llt, rootContainer)).GetEnumerator();
             for (; i < tries; i++)  
```

Comment:
Is this a high traffic performance critical section?

Reply:
Yes, hopefully fixed in https://github.com/ravendb/ravendb/pull/17047